drivers/espi/espi_mchp_xec.c

```diff
@@ -0,0 +1,822 @@
+	/* SMVW00 */
+	[ESPI_VWIRE_SIGNAL_OOB_RST_ACK]   = { SMVW00, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_WAKE]          = { SMVW00, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_PME]           = { SMVW00, SRC_ID3, S2M },
+	/* SMVW01 */
+	[ESPI_VWIRE_SIGNAL_SLV_BOOT_DONE] = { SMVW01, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_ERR_FATAL]     = { SMVW01, SRC_ID1, S2M },
+	[ESPI_VWIRE_SIGNAL_ERR_NON_FATAL] = { SMVW01, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_SLV_BOOT_STS]  = { SMVW01, SRC_ID3, S2M },
+	/* SMVW02 */
+	[ESPI_VWIRE_SIGNAL_SCI]           = { SMVW02, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_SMI]           = { SMVW02, SRC_ID1, S2M },
+	[ESPI_VWIRE_SIGNAL_RST_CPU_INIT]  = { SMVW02, SRC_ID2, S2M },
+	[ESPI_VWIRE_SIGNAL_HOST_RST_ACK]  = { SMVW02, SRC_ID3, S2M },
+	/* MSVW02 */
+	[ESPI_VWIRE_SIGNAL_HOST_RST_WARN] = { MSVW02, SRC_ID0, M2S },
+	/* SMVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_ACK]       = { SMVW03, SRC_ID0, S2M },
+	[ESPI_VWIRE_SIGNAL_DNX_ACK]       = { SMVW03, SRC_ID1, S2M },
+	/* MSVW03 */
+	[ESPI_VWIRE_SIGNAL_SUS_WARN]      = { MSVW03, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_SUS_PWRDN_ACK] = { MSVW03, SRC_ID1, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_A]         = { MSVW03, SRC_ID3, M2S },
+	/* MSVW04 */
+	[ESPI_VWIRE_SIGNAL_SLP_LAN]       = { MSVW04, SRC_ID0, M2S },
+	[ESPI_VWIRE_SIGNAL_SLP_WLAN]      = { MSVW04, SRC_ID1, M2S },
+	/* MSVW07 */
+	[ESPI_VWIRE_SIGNAL_HOST_C10]      = { MSVW07, SRC_ID0, M2S },
+	/* MSVW08 */
+	[ESPI_VWIRE_SIGNAL_DNX_WARN]      = { MSVW08, SRC_ID1, M2S },
+};
+
+static int espi_xec_configure(struct device *dev, struct espi_cfg *cfg)
+{
+	u8_t iomode = 0;
+	u8_t cap1 = ESPI_CAP_REGS->GLB_CAP1;
+	u8_t cur_iomode = (cap1 & MCHP_ESPI_GBL_CAP1_IO_MODE_MASK) >>
+		MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+
+	/* Set frequency */
+	cap1 &= ~MCHP_ESPI_GBL_CAP1_MAX_FREQ_MASK;
+	switch (cfg->max_freq) {
+	case 20:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_20M;
+		break;
+	case 25:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_25M;
+		break;
+	case 33:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_33M;
+		break;
+	case 50:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_50M;
+		break;
+	case 66:
+		cap1 |= MCHP_ESPI_GBL_CAP1_MAX_FREQ_66M;
+		break;
+	default:
+		return -EINVAL;
+	}
+
+	/* Set IO mode */
+	iomode = (iomode >> 1);
+	if (iomode > 3) {
+		return -EINVAL;
+	}
+
+	if (iomode != cur_iomode) {
+		cap1 &= ~MCHP_ESPI_GBL_CAP1_IO_MODE_MASK0 <<
+			MCHP_ESPI_GBL_CAP1_IO_MODE_POS;
+		cap1 |= (iomode << MCHP_ESPI_GBL_CAP1_IO_MODE_POS);
+	}
+	ESPI_CAP_REGS->GLB_CAP1 = cap1;
+
+	/* TODO: Consider to this later when configuring PC subdevices */
+	ESPI_HIO_BAR_REGS->IOBAR_INIT = MCHP_ESPI_IOBAR_INIT_DFLT;
+	ESPI_HIO_BAR_REGS->HOST_BAR_IOC |= MCHP_ESPI_IO_BAR_HOST_VALID;
+
+	/* Activate the eSPI block *.
+	 * Need to guarantee that this register is configured before RSMRST#
+	 * de-assertion and after pinmux
+	 */
+	ESPI_EIO_BAR_REGS->IO_ACTV = 1;
+	printk("eSPI block actived succesfully\n");
+	return 0;
+}
+
+static bool espi_xec_channel_ready(struct device *dev,
+	enum espi_channel ch)
```

Comment:
indentation should be below last '(' + 1 space. Same for function calls (as long as it fits the 80 chars). Apply that everywhere.